crates/key-value-azure/src/lib.rs

```diff
@@ -0,0 +1,164 @@
+use std::sync::Arc;
+
+use anyhow::Result;
+use azure_data_cosmos::{
+    prelude::{AuthorizationToken, CollectionClient, CosmosClient, Query},
+    CosmosEntity,
+};
+use futures::StreamExt;
+use serde::{Deserialize, Serialize};
+use spin_core::async_trait;
+use spin_key_value::{log_error, Error, Store, StoreManager};
+use tokio::{
+    sync::{Mutex, OnceCell},
+    task,
+};
+
+pub struct KeyValueAzureCosmos {
+    key: String,
+    account: String,
+    database: String,
+    collection: String,
+    pub client: OnceCell<Arc<Mutex<CollectionClient>>>,
+}
+
+impl KeyValueAzureCosmos {
+    pub fn new(key: String, account: String, database: String, collection: String) -> Result<Self> {
+        Ok(Self {
+            key,
+            account,
+            database,
+            collection,
```

Comment:
I do have a question about the terminology used by Azure CosmosDB with the NoSQL API this implementation is using: while the client SDK uses `collection`, the UI seems to show `container` as the name for the entity grouping the documents, and I'm wondering which one should we expose to the user.

Any thoughts, @Mossaka @devigned?

cc @itowlson

Reply:
I think container would be consistent with [the object model documentation](https://learn.microsoft.com/en-us/azure/cosmos-db/nosql/quickstart-dotnet?tabs=azure-portal%2Cwindows%2Cpasswordless%2Csign-in-azure-cli#object-model). Maybe collection was from old specs / docs to feel more MongoDB'ish.

On a more meta note, would specifying the containers as part of the configuration be limiting? Would a user likely use multiple containers or dynamically named containers?

Reply:
In wasi-keyvalue, the name used is `bucket`. https://github.com/WebAssembly/wasi-keyvalue/blob/main/wit/types.wit

Reply:
In this particular case I'd align with Cosmos nomenclature rather than wasi-keyvalue, because this is talking _specifically_ about the Cosmos back end store, not the WASI front end API.

Reply:
Updated this to be `container`.